src/state/controls/Button.ts

```diff
@@ -71,4 +71,11 @@ export class Button extends Control<IButtonData> implements IButton {
     public giveInput(input: IButtonInput): Promise<void> {
         return this.sendInput(input);
     }
+
+    public update(changedData: Partial<IButtonData>): Promise<void> {
+        if (changedData.cooldown) {
```

Comment:
I described the problem above. It allows developers(external, public facing) to update multiple values on the control at once returning one promise from that.

Its specifically targets the line above:
`control.setText('connor').then(() => control.setCooldown(5))` as not being a good experience.

It is intended to be the same level as those setters and used in the same context